portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -19,15 +22,43 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    UserService userService = UserServiceFactory.getUserService();
-    PrintWriter writer = response.getWriter();
 
     response.setContentType("text/html");
-    writer.println(userService.isUserLoggedIn(););
-  }
+    UserService userService = UserServiceFactory.getUserService();
 
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<String> loginInfo = new ArrayList<>();
+    loginInfo.add(String.valueOf(userService.isUserLoggedIn()));
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String logoutUrl = userService.createLogoutURL("/comments.html");
+        
+      loginInfo.add(logoutUrl);
+      loginInfo.add(getUserNickname(userService.getCurrentUser().getUserId(),
+          userService.getCurrentUser().getEmail()));
```

Comment:
[Optional] I've noticed you have a habit of doing all your definitions at the top followed by a blank line and then the logic.  Is there a particular reason for that?  It is generally preferred to defined variables as close as possible to where they are used and to only add additional newlines when you want to create a new conceptual grouping in the code.

Reply:
It's just something I got used to over my years of coding. However, what you said would definitely make more sense, so I will try to remember to do that more often.

Reply:
Makes sense.  I've seen other people do it and I think it's partly inspired by older languages where definitions had come at the start of functions.  It's not exactly wrong but I'm not a huge fan of the style either.